File: ios/NHSOnline/NHSOnline/Data/Enum/IntegrationLevel.swift
enum IntegrationLevel: String, Codable {
    case Gold
    case GoldOverlay
    case GoldWithNoHeaders
    case SilverWithoutWebNavigation
    case SilverWithWebNavigation
    case Bronze
    case Unknown

    init(from decoder: Decoder) throws {
        let label = try decoder.singleValueContainer().decode(String.self)

        self = IntegrationLevel(rawValue: label) ?? .Unknown

        if (self == .Unknown ) {
            Logger.logInfo(message: "An unknown IntegrationLevel enum has been encountered: %@", label)
        }
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/services/knownservices/enums/IntegrationLevel.kt
package com.nhs.online.nhsonline.services.knownservices.enums

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = false)
enum class IntegrationLevel {
    Gold,
    GoldOverlay,
    GoldWithNoHeaders,
    SilverWithoutWebNavigation,
    SilverWithWebNavigation,
    Bronze,
    Unknown,
}